### PR TITLE
Upgrade Hazelcast version to 5.1.1

### LINKED ISSUE #2611: Port Hazelcast upgrade to Gaffer 1.*
A PR #2599 was raised by dependabot because of a security vulnerability in older versions of Hazelcast. The dependabot PR couldn't be merged because migrating to the latest Hazelcast version requires minor code changes.

Changes to upgrade Hazelcast have already been made in the Gaffer 2 dependencies upgrade branch, these need to be ported to Gaffer 1.* to fix any potential security vulnerability.

### PR DESCRIPTION
**Pull request by GCHQDeveloper314**
# Related Issue

- Resolve #2611

### CODE REVIEW

File: library/cache-library/hazelcast-cache-service/src/test/resources/hazelcast.xml [hunk 1]
```diff
@@ -1,9 +1,6 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 <hazelcast xmlns="http://www.hazelcast.com/schema/config">
-    <group>
-        <name>test</name>
-        <password>password</password>
-    </group>
+    <cluster-name>test</cluster-name>
```
Comment (r32575) [inline library/cache-library/hazelcast-cache-service/src/test/resources/hazelcast.xml:3]: Is there still the option to have a user name/password?

Comment (GCHQDeveloper314) [reply library/cache-library/hazelcast-cache-service/src/test/resources/hazelcast.xml:3]: [According to this migration guide there is not](https://docs.hazelcast.com/imdg/4.2/migration-guides#changes-in-security-configurations), it appears to have been deprecated for a while.

**Comment by GCHQDeveloper314** [comment]
From what I can tell - I've only had a very quick look at the code - there's nothing for users to do as the code cannot connect to any external Hazelcast. I'm not sure why this module is called a 'cache service' at all. It's not what I would describe as a service, it's simply an implementation of `ICacheService` which uses a `HazelcastCache`. It doesn't offer any way to connect to an external Hazelcast cluster. It just [creates a single local 'node' inside the JVM and uses that](https://docs.hazelcast.org/docs/5.1/javadoc/com/hazelcast/core/Hazelcast.html#newHazelcastInstance-com.hazelcast.config.Config-).

[The code is very simple](https://github.com/gchq/Gaffer/tree/develop/library/cache-library/hazelcast-cache-service/src/main/java/uk/gov/gchq/gaffer/cache/impl) and this is why it is easy to migrate to this much newer version. I'm not sure this module is actually very useful to anyone as it is, it feels more like a proof of concept. Maybe we should consider deprecating it?

**Review by GCHQDev404 — APPROVED** [review]
No objections.

**Comment by GCHQDeveloper314** [comment]
After having a closer look at this, it seems that a cluster can in fact be joined by the node created by Gaffer - if this is [specified in the config](https://docs.hazelcast.com/hazelcast/latest/clusters/discovering-by-tcp). A user currently running a cluster using an old version of Hazelcast would need to upgrade their cluster to continue using it with Gaffer. In a cluster created by other instances of Gaffer, so long as the other instances were also updated, the only change needed would be updates to the Hazelcast config to be compatible with the new version.

**Review by r32575 — APPROVED** [review]

**Review by t92549 — APPROVED** [review]

**Pull request merged**